src/kOS/Communication/ConnectivityManager.cs

```diff
@@ -1,9 +1,146 @@
+            }
+        }
+
+        public static void RefreshInstance()
+        {
+            if (needEventInit)
+            {
+                // KSP's events don't support pointing to a static method, so we need to wrap the call
+                GameEvents.OnGameSettingsApplied.Add(() => { RefreshInstance(); });
+                GameEvents.onGameStatePostLoad.Add(value => { RefreshInstance(); });
+                needEventInit = false;
+            }
+            SafeHouse.Logger.LogError("ConnectivityManager.RefreshInstance()");
+            if (myinstance == null || myinstance.GetType() != GetSelectedManagerType())
+            {
+                SafeHouse.Logger.LogError("RefreshInstance - change manager");
+                myinstance = CreateManagerObject();
+                if (myinstance == null || !myinstance.IsEnabled)
+                {
+                    SafeHouse.Logger.LogError("ConnectivityManager.RefreshInstance - Failed to instantiate " + GetSelectedManagerType().Name);
+                    kOSConnectivityParameters.Instance.CheckNewManagers();
+                }
+            }
+        }
+
+        private static HashSet<Type> typeHash = new HashSet<Type>();
+
+        public static void RegisterMethod(Type t)
+        {
+            typeHash.Add(t);
+        }
+
+        public static List<string> GetStringList()
+        {
+            var ret = new List<string>();
+            foreach (var t in typeHash)
+            {
+                if (t == typeof(StockConnectivityManager))
+                {
+                    ret.Insert(0, t.Name);
+                }
+                else
+                {
+                    IConnectivityManager test = (IConnectivityManager)Activator.CreateInstance(t);
+                    if (test.IsEnabled)
+                    {
+                        ret.Add(t.Name);
+                    }
+                }
+            }
+            return ret;
+        }
+
+        public static HashSet<string> GetStringHash()
+        {
+            var ret = new HashSet<string>();
+            foreach (var t in typeHash)
+            {
+                IConnectivityManager test = (IConnectivityManager)Activator.CreateInstance(t);
+                if (test.IsEnabled)
+                {
+                    ret.Add(t.Name);
+                }
+            }
+            return ret;
+        }
+
+        public static Type GetSelectedManagerType()
+        {
+            string name = kOSConnectivityParameters.Instance.connectivityHandler;
+            return GetManagerType(name);
+        }
+
+        public static Type GetManagerType(string name)
+        {
+            foreach (var t in typeHash)
+            {
+                if (t.Name.Equals(name))
+                {
+                    return t;
+                }
+            }
+            return null;
+        }
+
+        public static IConnectivityManager CreateManagerObject()
+        {
+            var t = GetSelectedManagerType();
```

Comment:
Yeah, I tend to start with `var` mostly so I'm not strongly typed in case I make a change.  I didn't swap it back because the method mentions the word "Type" but I can strongly type it instead.